build/codegen/codegen.sh

```diff
@@ -14,29 +14,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+##GROUP_VERSIONS="ceph.rook.io:v1"
+
+scriptdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+basedir="$(cd "${scriptdir}/../.."  && pwd)"
+apisdir="${basedir}"/pkg/apis
+clientdir="${basedir}"/pkg/client
+
+boilerplate="${CODE_GENERATOR}/examples/hack/"boilerplate.go.txt
+kube_codegen="${CODE_GENERATOR}/kube_codegen.sh"
+
+
+echo "DEBUG: scriptdir: ${scriptdir}"
+echo "DEBUG: basedir: ${basedir1}"
+echo "DEBUG: apisdir: ${apisdir}"
+echo "DEBUG: clientdir: ${clientdir}"
+echo "DEBUG: CODE_GENERATOR: ${CODE_GENERATOR}"
+echo "DEBUG: boilerplate ${boilerplate}"
+echo "DEBUG: kube_codegen: ${kube_codegen}"
+ echo "DEBUG: done"
+# exit 0
+
+
+
+set -o pipefail
```

Comment:
Per the codegen example file, please use this pattern and location within the file: 
https://github.com/kubernetes-sigs/container-object-storage-interface/blob/7ddc93baaa3f08c9c8990a17c7b958955d93c044/client/hack/update-codegen.sh#L17C1-L19C16

This will also mean changing the shebang to `#!/usr/bin/env bash` without the `-S` or `-e`

Reply:
done.

I wanted to do the shebang change anyway, since without it, `make codegen`does not run at all in my local ubuntu tests.